src/main/scala/encry/consensus/PowLinearController.scala

```diff
@@ -10,8 +10,12 @@ object PowLinearController {
 
   private val chainParams = Constants.Chain
 
-  def getDifficulty(previousHeaders: Seq[(Int, EncryBlockHeader)]): NBits = {
-    if (previousHeaders.length == chainParams.RetargetingEpochsQty) {
+  val PrecisionConstant: Int = 1000000000
+
+  def getDifficulty(previousHeaders: Seq[(Int, EncryBlockHeader)]): NBits =
+    if (previousHeaders.lengthCompare(1) == 0 || previousHeaders.head._2.timestamp >= previousHeaders.last._2.timestamp)
```

Comment:
if `previousHeaders.lengthCompare(1) == 0` is true, then second condition also will true
if `previousHeaders` is empty, then unhandled exception will arise

Reply:
Empty collection could not be passed to this function.